### ISSUE TITLE
Multiple choices enum is not correctly reflected in filters

### ISSUE THREAD

**Issue by magrinj** [supabase/pg_graphql#458]
**Describe the bug**

Simple choice enums are correctly generated and can be filtered, but not multiple choices enums.

**To Reproduce**
1. Create a multiple choice enum in the Postgres database:
```sql
CREATE TYPE document_multiSelect_enum AS ENUM('MULTI_FIRST','MULTI_SECOND');
```
2. Create a table using this enum:
```sql
CREATE TABLE document (id serial, multiSelect document_multiSelect_enum[]);
```
3. Insert a new record:
```sql
INSERT INTO document(document_multiSelect_enum) VALUES('{MULTI_FIRST, MULTI_SECOND}');
```
4. Try to query this document with a filter:
```sql
SELECT graphql.resolve($$
      query {
        documentCollection(filter: {multiSelect:{eq:"MULTI_FIRST"}}) {
          edges {
            node {
              id
              multiSelect
            }
          }
        }
      }
      $$);
```
5. This error occurred:
```json
{"data": null, "errors": [{"message": "Input for type documentFilter contains extra keys [\"multiSelect\"]"}]}
```
6. In the graphQL introspection, `document_multiselect_enumFilter` of kind `INPUT_OBJECT` exist but is not inside `documentFilter` of kind `INPUT_OBJECT` as it should. It actually works properly for a single choice enum.

**Expected behavior**

The result should be properly filtered.

**Versions:**
 - PostgreSQL: 14.9.1
 - pg_graphql: 1.4.2

**Comment by olirice** [comment]
I made a few edits to your example so we can discuss:
```sql
create type "my_enum" as enum('MULTI_FIRST','MULTI_SECOND');

create table doc (
	id serial primary key,
	"enumArr" my_enum[],
	"otherArr" int[] default '{1,2}'
);

insert into doc("enumArr") values ('{MULTI_FIRST, MULTI_SECOND}');
```

Some snippets from the resulting GraphQL schema
```graphql
type Doc implements Node {
  """Globally Unique Record Identifier"""
  nodeId: ID!
  id: Int!
  enumArr: [MyEnum]
  otherArr: [Int]
}

input DocFilter {
  id: IntFilter
  nodeId: IDFilter

  """
  Returns true only if all its inner filters are true, otherwise returns false
  """
  and: [DocFilter!]

  """
  Returns true if at least one of its inner filters is true, otherwise returns false
  """
  or: [DocFilter!]

  """Negates a filter"""
  not: DocFilter
}

type Query {
  """A pagable collection of type `Doc`"""
  docCollection(
    """Query the first `n` records in the collection"""
    first: Int

    """Query the last `n` records in the collection"""
    last: Int

    """Query values in the collection before the provided cursor"""
    before: Cursor

    """Query values in the collection after the provided cursor"""
    after: Cursor

    """Filters to apply to the results set when querying from the collection"""
    filter: DocFilter

    """Sort order to apply to the collection"""
    orderBy: [DocOrderBy!]
  ): DocConnection
}
```

You can see from `DocFilter` that only the `id` and `nodeId` are filterable. That's because pg_graphql doesn't support filter operations on array types. This isn't limited to just enums, it is identical for the dummy integer array field named `otherArr`.

---

I'm going to close this issue since it might be a confusing starting place given the emphasis on enums, but If you'd like to open an issue requesting support for filtering arrays I think that would be well received.

We've had multiple community contributions to filtering the past and would be a good first issue if a team member doesn't grab it.

We'd probably want support for `eq` and `contains`

**Comment by olirice** [comment]
Could you give me some more info on how you're using `pg_graphql` in twenty?

From your [docs](https://docs.twenty.com/graphql/) it doesn't appear to be pg_graphql based. Is that an ongoing transition, unrelated, etc?

If you're transitioning to it and run into any trouble, or just want to chat, please let me know. I'd love to hear more about what you're doing!

**Comment by magrinj** [comment]
Hello @olirice,

Thank you for your detailed explanations. Based on your insights, I plan to initiate a new ticket to incorporate arrays support.

At [Twenty](https://github.com/twentyhq/twenty), our approach to utilizing `pg_graphql` is somewhat akin to an ORM, primarily for our generative backend operations. This setup allows our clients to create custom objects or add custom fields to standard objects. To facilitate this, we maintain metadata that describes these objects and columns. Utilizing this metadata, we leverage TypeORM to generate migrations for creating tables and columns accordingly.

We ensure data segregation for each client workspace by splitting them into separate schemas within the database. Additionally, some core entities are constructed using Nest.JS. These are then integrated with a custom GraphQL schema, developed in-house, leveraging Yoga's `conditionalSchema` feature.

In essence, `pg_graphql` plays a crucial role in [querying the database](https://github.com/twentyhq/twenty/tree/main/server/src/workspace/workspace-query-builder). It does so by interfacing with the schema generated from our end, making database queries more streamlined and efficient, particularly when aligned with `GraphQLResolveInfo`.

If you're interested, I'd be happy to discuss this further. We can continue our conversation on our [Discord](https://discord.gg/cx5n4Jzs57), or I can join you on your preferred platform, be it Slack, Discord, or any other. Let me know what works best for you!

**Comment by magrinj** [comment]
For reference, here the new ticket #460
